src/mavedb/routers/score_sets.py

```diff
@@ -1241,24 +1254,22 @@ async def update_score_set(
         # re-validate existing variants and clear them if they do not pass validation
         if item.variants:
             assert item.dataset_columns is not None
-            score_columns = [
-                "hgvs_nt",
-                "hgvs_splice",
-                "hgvs_pro",
-            ] + item.dataset_columns["score_columns"]
-            count_columns = [
-                "hgvs_nt",
-                "hgvs_splice",
-                "hgvs_pro",
-            ] + item.dataset_columns["count_columns"]
+            score_columns = {
+                "core": ["hgvs_nt", "hgvs_splice", "hgvs_pro"],
+                "mavedb": item.dataset_columns["score_columns"],
+            }
+            count_columns = {
+                "core": ["hgvs_nt", "hgvs_splice", "hgvs_pro"],
+                "mavedb": item.dataset_columns["count_columns"],
+            }
 
             scores_data = pd.DataFrame(
-                variants_to_csv_rows(item.variants, columns=score_columns, dtype="score_data")
+                variants_to_csv_rows(item.variants, columns=score_columns)
             ).replace("NA", pd.NA)
 
             if item.dataset_columns["count_columns"]:
                 count_data = pd.DataFrame(
-                    variants_to_csv_rows(item.variants, columns=count_columns, dtype="count_data")
+                    variants_to_csv_rows(item.variants, columns=count_columns)
```

Comment:
I think we will need to change how this works slightly. Probably the easiest way is to add another parameter to the `variants_to_csv_row` function `namespaced=True`. As written, the re-validated data will be stored with namespaces when we want it to be stored with just the column names.

If we add the `namespaced` parameter to the `variants_to_csv_rows` function, we can pass it `namespaced=False` in this context so that it doesn't namespace any of the column names and then doesn't save those columns with the namespace. In other contexts, we will get the namespaces back.

We'd need to add some additional logic to the `variants_to_csv_rows` to handle the new parameter, but that would be pretty straightforward.

Reply:
I'll modify this part next week. Branch release-2025.5.0 has some conflict parts.